### PR TITLE
[sparkle] - enh: `Button`

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR enhances the `Button` component in sparkle, aligning it with recent UI improvements and standardization efforts. 

The changes streamline the component's implementation while maintaining its existing functionality and introducing new features. Key changes include:
- Introducing new variants: "highlight", "warning", "outline", and "ghost" to replace the old "primary", "secondary", and "tertiary" variants
- Implementing a new "isLoading" state with an integrated Spinner component
- Adding support for tooltips and dropdown functionality
- Updating the component's styling to match the new design system

## Risk

Quite high blast radius in terms of UX as the button component is used in many places

## Deploy Plan

- Publish sparkle release candidate
- Deploy front edge
- Publish sparkle
- Deploy front

### CODE REVIEW

File: sparkle/src/components/Tree.tsx [hunk 1]
```diff
@@ -161,7 +161,7 @@ Tree.Item = function ({
           <IconButton
             icon={isExpanded ? ArrowDownSIcon : ArrowRightSIcon}
             size="xs"
-            variant="secondary"
+            variant="highlight"
```
Comment (flvndvd) [inline sparkle/src/components/Tree.tsx:164]: Why do we sometimes replace `secondary` by ghost and some other times by `highlight`?

File: sparkle/src/components/Button.tsx [hunk 2]
```diff
@@ -1,261 +1,187 @@
...
+  primary: "light",
+  highlight: "light",
+  warning: "light",
+  outline: "slate400",
+  ghost: "slate400",
+};
 
-const magnifyingClasses =
-  "hover:s-scale-105 hover:s-drop-shadow-md active:s-scale-100 active:s-drop-shadow-none";
+export interface MetaButtonProps
```
Comment (flvndvd) [inline sparkle/src/components/Button.tsx:80]: We probably don't want to export this interface.

**Review by flvndvd — COMMENTED** [review]
Just checked out the branch, and it looks like the size variants don't work in there, expected?

File: sparkle/src/components/Tree.tsx (hunk 1 above)
Comment (JulesBelveze) [reply sparkle/src/components/Tree.tsx:164]: Yeah I didn't really bother for consistency in the stories but "secondary" should be "highlight". I will make it consistant

File: sparkle/src/components/Button.tsx (hunk 2 above)
Comment (JulesBelveze) [reply sparkle/src/components/Button.tsx:80]: Indeed, mistake

File: sparkle/src/components/Tree.tsx (hunk 1 above)
Comment (JulesBelveze) [reply sparkle/src/components/Tree.tsx:164]: Should be consistant now

**Review by flvndvd — APPROVED** [review]

**Pull request merged**